### ISSUE TITLE
Slim3 middleware on route groups doesn't work

### ISSUE THREAD

**Issue by w4zZz4p** [slimphp/Slim#1377]
The problem exists when adding middleware to route groups. Example below:

```
class Test {
    public function run($request, $response, $next){
        $response->write('+');
        $response = $next($request, $response);
        $response->write('-');
        return $response;
    }
}
```
#### Fails

```
$app = new \Slim\App();
$app->group('/admin', function () use ($app) {
    $app->get('/test', function($request, $response, $args) {
        return $response->write('works!');
    });
})->add('Test:run');
$app->run();
```
#### Works

But if you add same middleware to single route everything works fine.

```
$app = new \Slim\App();
$app->group('/admin', function () use ($app) {
    $app->get('/test', function($request, $response, $args) {
        return $response->write('works!');
    })->add('Test:run');
});
$app->run();
```

**Comment by silentworks** [comment]
You are correct, this seem to be a bug. I noticed the group doesn't take into consideration anything attached to the group when using the Container to resolve the middleware callable, this however works on a per route level.

//cc @opengeek

**Comment by akrabat** [comment]
Fixed in #1381
